Formula/s/sonar-scanner.rb

```diff
@@ -20,7 +20,7 @@ def install
     ln_s etc/"sonar-scanner.properties", libexec/"conf/sonar-scanner.properties"
     bin.env_script_all_files libexec/"bin/",
                               SONAR_SCANNER_HOME: libexec,
-                              JAVA_HOME:          Formula["openjdk"].opt_prefix
+                              JAVA_HOME:          Formula["openjdk@17"].opt_prefix
```

Comment:
This should let you use whichever Java version you like by setting `JAVA_HOME`:
```suggestion
                              JAVA_HOME:          Language::Java.overridable_java_home_env
```
But this requires dropping the change to `depends_on`.